fuzz/src/lib.rs

```diff
@@ -180,6 +221,64 @@ impl Creation<'_> {
                     _ => panic!("CompactStr and core::str read UTF-8 differently?"),
                 }
             }
+            NumToCompactStr(num_type) => {
+                // Create a `CompactStr` and `String` from a number
+                let (compact, word) = match num_type {
+                    NumType::U8(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::I8(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::U16(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::I16(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::U32(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::I32(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::U64(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::I64(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::U128(val) => (val.to_compact_str(), val.to_string()),
+                    NumType::I128(val) => (val.to_compact_str(), val.to_string()),
+                    // NumType::F32(val) => (val.to_compact_str(), val.to_string()),
+                    // NumType::F64(val) => (val.to_compact_str(), val.to_string()),
+                };
+
+                assert_eq!(compact, word);
+
+                // Numbers should always be inlined, except 64 or 128 bit integers, they might be
+                // heap allocated
+                match num_type {
```

Comment:
I think it is possible to do this:

```suggestion
                match num_type {
                    NumType::U128(_) | NumType::I128(_) => {
                        assert_properly_allocated(&compact, &word);
                    }
                    #[cfg(target_pointer_width = "32")]
                    NumType::U64(_) | NumType::I64(_) => {
                        assert_properly_allocated(&compact, &word);
                    }
                    _ => assert!(!compact.is_heap_allocated()),
                };
```

Reply:
Changed this to a different general case